src/NuGet.ApplicationInsights.Owin/OwinRequestIdContext.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.Remoting.Messaging;
+
+namespace NuGet.ApplicationInsights.Owin
+{
+    public static class OwinRequestIdContext
+    {
+        public static void Set(string value)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                CallContext.LogicalSetData(ApplicationInsightsMiddleware.OwinRequestIdKey, value);
+            }
+        }
```

Comment:
That makes sense. I totally missed that point of initializing the id. So, all the telemetry transmitted within the scope of this request will have the same operationId. Thanks!